Plutarch/DataRepr.hs

```diff
@@ -136,3 +146,36 @@ instance (Ledger.FromData h, Ledger.ToData h, PIsData p) => PLift (PIsDataReprIn
     maybeToRight "Failed to decode data" $ Ledger.fromData h
     where
       maybeToRight e = maybe (Left e) Right
+
+{- | 
+  DerivingVia wrapper for deriving `PLift` instances
+  via the wrapped type, while lifting to a coercible Haskell type.
+
+-}
+newtype PVia (p :: k -> Type) (h :: Type) (s :: k) = PVia (Term s p)
```

Comment:
If this is used to derive only `PLift` instances, I suggest putting it in `Plutarch.Lift` module, alongside `PBuiltinType` - because it looks like they serve, essentially, the same purpose (with the only difference being that `PVia` is used to derive new types, and `PBuiltinType` is used to derive builtin types).

I also wonder why it is named `PVia` (rather than say `PNewType` to be consistent with `PBuiltinType` naming).

Reply:
Good point, yeah. I'll move it over.

I'd say `PNewType` is less accurate than `PVia`, since it uses `Coercible`, it's a bit more powerful than just newtypes (same deal with deriving via vs deriving newtype, really).

Reply:
Actually, I'll keep `PVia` where it is, to avoid the module cycle due to the `PlutusType` instance.